### PR TITLE
fix(ecs): configFileValue on FirelensOptions should be an optional property

### PR DESCRIPTION
**Pull request by david-doyle-as24**
----

Firelens config mistakenly requires configFileType, when [this documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-firelensconfiguration.html) allows that any of firelens config options can be set or unset. This PR fixes this bug and sets the "enable-ecs-log-metadata" field by default to true, even when the options field is not set.

Closes [#16594](https://github.com/aws/aws-cdk/issues/16594)

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ecs/lib/firelens-log-router.ts [hunk 1]
```diff
@@ -78,7 +79,7 @@ export interface FirelensConfig {
 
   /**
    * Firelens options
-   * @default - no additional options
+   * @default - EnableECSLogMetadata option is set to true
```
Comment (upparekh) [inline packages/@aws-cdk/aws-ecs/lib/firelens-log-router.ts:82]: nit: `@default - ECS log metadata enabled`

File: packages/@aws-cdk/aws-ecs/lib/firelens-log-router.ts [hunk 2]
```diff
@@ -200,37 +201,43 @@ export class FirelensLogRouter extends ContainerDefinition {
...
         props.taskDefinition.addToExecutionRolePolicy(new iam.PolicyStatement({
           actions: [
             's3:GetBucketLocation',
           ],
-          resources: [options.configFileValue.split('/')[0]],
+          resources: [options?.configFileValue.split('/')[0]],
         }));
       }
+    } else if (options?.configFileType && !options?.configFileValue) {
+      throw new Error('Firelens Config Error: Config file value cannot be undefined when config file type is declared.');
```
Comment (upparekh) [inline packages/@aws-cdk/aws-ecs/lib/firelens-log-router.ts:233]: Can we also add a unit test to assert this error being thrown?

File: packages/@aws-cdk/aws-ecs/lib/firelens-log-router.ts [hunk 3]
```diff
@@ -116,7 +117,7 @@ function renderFirelensConfig(firelensConfig: FirelensConfig): CfnTaskDefinition
       options: {
```
Comment (upparekh) [inline packages/@aws-cdk/aws-ecs/lib/firelens-log-router.ts:117]: I think we can also remove the `if-else` condition here to always return the firelens config in the `else` part (as the `enable-ecs-log-metadata` in the `options` will always be present now).

**Review by upparekh — COMMENTED** [review]
Hi @david-doyle-as24 , awesome PR! Thanks for working on this.

I just have two small suggestions!

**Comment by upparekh** [comment]
It seems from the Build Logs like the `integ.all-service-addons.js` test is failing. You need to update the test as well.

**Comment by madeline-k** [comment]
Hello, I am closing this PR since it is in a build-failing state, and has not been touched in a long time. If anyone is still interested to contribute a fix here, please open a new PR.

**Pull request closed without merging**